src/main/java/seedu/address/logic/commands/AttendanceCommand.java

```diff
@@ -0,0 +1,215 @@
+
+    public static final String MESSAGE_ATTENDANCE_SUCCESS = "Attendance updated: %1$s";
+    public static final String MESSAGE_PET_NOT_ADDED = "Seems like this pet is not in the address book yet! "
+                                                        + "Make sure to add the pet in first!";
+
+    private final Index index;
+    private final PetAttendanceDescriptor attendanceDescriptor;
+
+
+    /**
+     * @param index of the pet in the filtered pet list to edit
+     * @param attendanceDescriptor details of attendance to edit the pet with
+     */
+    public AttendanceCommand(Index index, PetAttendanceDescriptor attendanceDescriptor) {
+        requireNonNull(index);
+        requireNonNull(attendanceDescriptor);
+
+        this.index = index;
+        this.attendanceDescriptor = new PetAttendanceDescriptor(attendanceDescriptor);
+    }
+
+    @Override
+    public CommandResult execute(Model model) throws CommandException {
+        requireNonNull(model);
+        List<Pet> lastShownList = model.getFilteredPetList();
+
+        if (index.getZeroBased() >= lastShownList.size()) {
+            throw new CommandException(Messages.MESSAGE_INVALID_PET_DISPLAYED_INDEX);
+        }
+
+        Pet petToUpdateAttendance = lastShownList.get(index.getZeroBased());
+        createAttendanceForPet(petToUpdateAttendance, attendanceDescriptor);
+
+        return new CommandResult(MESSAGE_ATTENDANCE_SUCCESS);
+    }
+
+    /**
+     * Creates an entry in individual pet's JSON file and txt file.
+     */
+    private static void createAttendanceForPet(Pet petToEdit,
+                           AttendanceCommand.PetAttendanceDescriptor petAttendanceDescriptor) throws CommandException {
+        assert petToEdit != null;
+
+        Name nameOfPet = petToEdit.getName();
+        Phone phoneNumberOfOwner = petToEdit.getPhone();
+        String jsonFilePath = "data/pets/"
+                + nameOfPet.toString().replaceAll("\\s", "")
+                + phoneNumberOfOwner.toString()
+                + ".json";
+        String txtFilePath = "data/pets/"
+                + nameOfPet.toString().replaceAll("\\s", "")
+                + phoneNumberOfOwner.toString()
+                + ".txt";
+        String attendanceDate = petAttendanceDescriptor.getAttendanceDateString();
+        String pickUpTime = petAttendanceDescriptor.getPickUpTimeString();
+        String dropOffTime = petAttendanceDescriptor.getDropOffTimeString();
+
+        try {
+            File file = new File(txtFilePath);
+            FileWriter fileWriter = new FileWriter(file, true);
+            fileWriter.write(attendanceDate + " " + pickUpTime + " " + dropOffTime + "\n");
+            fileWriter.close();
+
+            ArrayList<String> sortAttendance = new ArrayList<>();
+            Scanner fileScanner = new Scanner(file);
+            while (fileScanner.hasNextLine()) {
+                String lineAttendance = fileScanner.nextLine();
+                sortAttendance.add(lineAttendance);
```

Comment:
Possible to have this attendance list stored in the individual pets for easier access?